Mathlib/CategoryTheory/EffectiveEpi/Comp.lean

```diff
@@ -19,37 +19,106 @@ open Limits
 variable {C : Type*} [Category C]
 
 /--
-A split epi followed by an effective epi is an effective epi. This version takes an explicit section
-to the split epi, and is mainly used to define `effectiveEpiStructCompOfEffectiveEpiSplitEpi`,
+An effective epi family precomposed by a family of split epis is effective epimorphic.
+This version takes an explicit section to the split epis, and is mainly used to define
+`effectiveEpiStructCompOfEffectiveEpiSplitEpi`,
 which takes a `IsSplitEpi` instance instead.
 -/
 noncomputable
-def effectiveEpiStructCompOfEffectiveEpiSplitEpi' {B X Y : C} (f : X ⟶ B) (g : Y ⟶ X) (i : X ⟶ Y)
-    (hi : i ≫ g = 𝟙 _) [EffectiveEpi f] : EffectiveEpiStruct (g ≫ f) where
-  desc e w := EffectiveEpi.desc f (i ≫ e) fun g₁ g₂ _ ↦ (by
+def effectiveEpiFamilyStructCompOfEffectiveEpiSplitEpi' {α : Type*} {B : C} {X Y : α → C}
+    (f : (a : α) → X a ⟶ B) (g : (a : α) → Y a ⟶ X a) (i : (a : α) → X a ⟶ Y a)
+    (hi : ∀ a, i a ≫ g a = 𝟙 _) [EffectiveEpiFamily _ f] :
+    EffectiveEpiFamilyStruct _ (fun a ↦ g a ≫ f a) where
+  desc e w := EffectiveEpiFamily.desc _ f (fun a ↦ i a ≫ e a) fun a₁ a₂ g₁ g₂ _ ↦ (by
     simp only [← Category.assoc]
-    apply w (g₁ ≫ i) (g₂ ≫ i)
+    apply w _ _ (g₁ ≫ i a₁) (g₂ ≫ i a₂)
     simpa [← Category.assoc, hi])
-  fac e w := by
-    simp only [Category.assoc, EffectiveEpi.fac]
-    rw [← Category.id_comp e, ← Category.assoc, ← Category.assoc]
+  fac e w a := by
+    simp only [Category.assoc, EffectiveEpiFamily.fac]
+    rw [← Category.id_comp (e a), ← Category.assoc, ← Category.assoc]
     apply w
     simp only [Category.comp_id, Category.id_comp, ← Category.assoc]
     aesop
   uniq _ _ _ hm := by
-    apply EffectiveEpi.uniq f
-    rw [← hm, ← Category.assoc, ← Category.assoc, hi, Category.id_comp]
+    apply EffectiveEpiFamily.uniq _ f
+    intro a
+    rw [← hm a, ← Category.assoc, ← Category.assoc, hi, Category.id_comp]
 
-/-- A split epi followed by an effective epi is an effective epi. -/
+/--
+An effective epi family precomposed with a family of split epis is effective epimorphic.
+-/
 noncomputable
-def effectiveEpiStructCompOfEffectiveEpiSplitEpi {B X Y : C} (f : X ⟶ B) (g : Y ⟶ X) [IsSplitEpi g]
-    [EffectiveEpi f] : EffectiveEpiStruct (g ≫ f) :=
-  effectiveEpiStructCompOfEffectiveEpiSplitEpi' f g
-    (IsSplitEpi.exists_splitEpi (f := g)).some.section_
-    (IsSplitEpi.exists_splitEpi (f := g)).some.id
+def effectiveEpiFamilyStructCompOfEffectiveEpiSplitEpi {α : Type*} {B : C} {X Y : α → C}
+    (f : (a : α) → X a ⟶ B) (g : (a : α) → Y a ⟶ X a) [∀ a, IsSplitEpi (g a)]
+    [EffectiveEpiFamily _ f] : EffectiveEpiFamilyStruct _ (fun a ↦ g a ≫ f a) :=
+  effectiveEpiFamilyStructCompOfEffectiveEpiSplitEpi' f g
+    (fun a ↦ (IsSplitEpi.exists_splitEpi (f := g a)).some.section_)
```

Comment:
Here (and below), you should use the `IsSplitEpi` API: `section_` and the lemma `IsSplitEpi.id`.